src/R/Components/Impl/ConnectionManager/Implementation/Connection.cs

```diff
@@ -20,5 +30,43 @@ internal class Connection : ConnectionInfo, IConnection {
         /// If true, the connection is to a remote machine
         /// </summary>
         public bool IsRemote { get; }
+
+        public override string UserProvidedPath {
+            get { return base.UserProvidedPath; }
+            set {
+                base.UserProvidedPath = value;
+                Path = ToCompletePath(value);
+            }
+        }
+
+        public static string ToCompletePath(string path) {
```

Comment:
Similar to, say, e-mail connections: user cares about e-mail and password but doesn't care about POP or IMAP details added internally.